### PR TITLE
Fix shipping method display issue

### PR DESCRIPTION
**Pull request by nimish13**
Issue -
1. Wrong logic use in finding which shipping method should be displayed on frontend and backend
2. Method marked to be displayed on frontend only is also available on backend.
3. ``DISPLAY_ON_BACK_END`` constant is never being used.
4. Method that should be private were written as public method
5. avialble_to display should be a predicate method.

Fix.
1) Make separate method to check which Shipping Method should be available on frontend and backend.
2) Use ``DISPLAY_ON_BACK_END`` instead of ``DISPLAY_ON_FRONT_AND_BACK_END`` to ensure only shipping methods with display_on in [both, backend] are available on backend
3) Simplify logic of ``avialble_to display`` method and make it a predicate method.

### CODE REVIEW

File: core/spec/models/spree/shipping_method_spec.rb [hunk 1]
```diff
@@ -92,4 +95,31 @@ class DummyShippingCalculator < Spree::ShippingCalculator
       expect(shipping_method.deleted_at).not_to be_blank
     end
   end
+
+  describe '#available_to_display?' do
+    context 'when available on frontend' do
+      it { expect(frontend_shipping_method.available_to_display?(Spree::ShippingMethod::DISPLAY_ON_FRONT_END)).to be_truthy }
```
Comment (krzysiek1507) [inline core/spec/models/spree/shipping_method_spec.rb]: `be_truthy`? So method `available_to_display ` can return something other than `false|true`?

Comment (nimish13) [inline core/spec/models/spree/shipping_method_spec.rb]: @krzysiek1507 No this method will always return true | false. To be_truthy can also be used for the true check.
But still, for better understanding, I will change this ``to be_true``

Comment (nimish13) [inline core/spec/models/spree/shipping_method_spec.rb]: @krzysiek1507 I had made the changes. Please have a look at them.

**Comment by tanmay3011** [comment]
@nimish13 Please squash commits

**Comment by nimish13** [comment]
@tanmay3011 Single commit created for all three. :)

**Review by nnande — APPROVED** [review]

**Comment by krzysiek1507** [comment]
Should we backport it to 3.2, 3.1 and 3.0?

**Pull request merged**